src/main/java/de/tum/in/www1/artemis/service/ComplaintResponseService.java

```diff
@@ -43,51 +42,211 @@ public ComplaintResponseService(ComplaintRepository complaintRepository, Complai
+        User user = this.userService.getUser();
+        StudentParticipation studentParticipation = (StudentParticipation) originalResult.getParticipation();
+        if (!isUserAuthorizedToRespondToComplaint(studentParticipation, complaintFromDatabase, assessor, user)) {
+            throw new AccessForbiddenException("Insufficient permission for removing the lock on the complaint");
+        }
+        // only instructors and the original reviewer can remove the lock while it is still running
+        if (blockedByLock(unfinishedComplaintResponse, user, studentParticipation)) {
+            throw new ComplaintResponseLockedException(unfinishedComplaintResponse);
+        }
+        complaintResponseRepository.deleteById(unfinishedComplaintResponse.getId());
+        log.debug("Removed empty complaint and thus lock for complaint with id : {}", complaintId);
+    }
 
-        // Do not trust user input
-        Optional<Complaint> originalComplaintOptional = complaintRepository.findByIdWithEagerAssessor(complaintId);
-        if (originalComplaintOptional.isEmpty()) {
-            throw new BadRequestAlertException("The complaint you are referring to does not exist", ENTITY_NAME, "noresult");
+    private ComplaintResponse getUnfinishedComplaintResponse(Complaint complaintFromDatabase) {
+        if (complaintFromDatabase.isAccepted() != null) {
+            throw new IllegalArgumentException("Complaint is already handled");
         }
+        if (complaintFromDatabase.getComplaintResponse() == null) {
+            throw new IllegalArgumentException("Complaint response does not exists for given complaint");
+        }
+        if (complaintFromDatabase.getComplaintResponse().getSubmittedTime() != null) {
+            throw new IllegalArgumentException("Complaint response is already submitted");
+        }
+        return complaintFromDatabase.getComplaintResponse();
+    }
 
-        Complaint originalComplaint = originalComplaintOptional.get();
+    /**
+     * Refreshes the empty complaint response for a given complaint that acts a lock
+     *
+     * The empty complaint response acts as a lock. Only the reviewer of the empty complaint response and instructors can resolve the complaint as long as the lock
+     * is running. For lock duration calculation see: {@link ComplaintResponse#isCurrentlyLocked()}. This methods exchanges the current empty complaint response to a new one
+     * thus updating the lock.
+     *
+     * This is possible in two cases:
+     *
+     * Case A: Lock is currently active -> Only the initial reviewer or an instructor can refresh the empty complaint response
+     * Case B: Lock has run out --> Others teaching assistants can refresh the empty complaint response thus acquiring the lock
+     *
+     * @param complaintId if of the complaint for which to refresh the empty response for
+     * @return refreshed empty complaint response
+     */
+    @Transactional // ok because of modifying query
```

Comment:
please remove `@Transactional`

Reply:
Done in 8e3e3774528e1dc63f6450e98a65532ae01f7f06